backends/dpdk/dpdkContext.cpp

```diff
@@ -200,6 +188,33 @@ Util::JsonObject* DpdkContextGenerator::initTableCommonJson(const cstring name,
     return tableJson;
 }
 
+void DpdkContextGenerator::collectHandleId() {
+    for (auto table : p4info.tables()) {
+        const auto& pre_t = table.preamble();
+        context_handle_map[pre_t.name().c_str()] = pre_t.id();
+        for (auto& action_ref : table.action_refs()) {
+            auto* action = P4::BFRT::Standard::findAction(p4info, action_ref.id());
+            if (action == nullptr) {
+                ::error(ErrorType::ERR_INVALID, "Invalid action id '%1%'", action_ref.id());
+                continue;
+            }
+            const auto& pre_a = action->preamble();
+            context_handle_map[pre_a.name()] = pre_a.id();
+        }
+    }
+}
+
+size_t DpdkContextGenerator::getHandleId(cstring name) {
+    size_t id = 0;
+    for (auto x : context_handle_map) {
+        if (x.first.find(name.c_str())) {
+            id = context_handle_map[x.first];
+            break;
+        }
+    }
+    return id;
```

Comment:
Is it ok to return 0 if handle is not found? 
IMO, either fallback to older method or asserting would be fine.

Reply:
Yes I think asserting would be better.

Reply:
added bugcheck